commons/src/test/java/org/eclipse/kapua/commons/configuration/metatype/TmetadataImplTest.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.commons.configuration.metatype;
+
+import org.eclipse.kapua.model.config.metatype.KapuaTdesignate;
+import org.eclipse.kapua.model.config.metatype.KapuaTocd;
+import org.eclipse.kapua.qa.markers.junit.JUnitTests;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import javax.xml.namespace.QName;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Category(JUnitTests.class)
+public class TmetadataImplTest extends Assert {
+
+    @Before
+    public void createInstanceOfClasses() {
+        emptyTocd = new EmptyTocd();
+        tocd = new TocdImpl();
+        tmetadata = new TmetadataImpl();
+        tdesignate = new TdesignateImpl();
+    }
+
+    EmptyTocd emptyTocd;
+    TocdImpl tocd;
+    TmetadataImpl tmetadata;
+    TdesignateImpl tdesignate;
+
+    @Test
+    public void getOCDTest() {
+        assertTrue(tmetadata.getOCD().isEmpty());
+    }
+
+    @Test
+    public void setAndGetOCDTest() {
+        List<KapuaTocd> ocd = new ArrayList<>();
+        ocd.add(tocd);
+        ocd.add(emptyTocd);
+        tmetadata.setOCD(ocd);
+        assertEquals(ocd, tmetadata.getOCD());
+    }
+
+    @Test
+    public void getDesignateTest() {
+        assertTrue(tmetadata.getDesignate().isEmpty());
+    }
+
+    @Test
+    public void setAndGetDesignateTest() {
+        List<KapuaTdesignate> designate = new ArrayList<>();
+        designate.add(tdesignate);
+        tmetadata.setDesignate(designate);
+        assertEquals(designate, tmetadata.getDesignate());
```

Comment:
While using `assertEquals`, `assertNotEquals` (and so on) they al support a message linked to the `assert*` as a first argument. It is a good practice to add a message that shows which specific assert has failed.

So in this case:
`assertEquals("tmetadata.designate", designate, tmetadata.getDesignate());`
would make any failure linked to a meaningful message which can avoid to have to look at the code to check what has failed.